### PR TITLE
try fixing readdocs by fixing structure

### PR DESCRIPTION
**Pull request by chiaral**
I think the api now doesn't work because the structure didn't have src in it

### CODE REVIEW

**Comment by chiaral** [comment]
nope didn't work!

**Comment by chiaral** [comment]
I think this is ready to merge - I have a queued Travis CI does it matter @rabernat ?

**Review by chiaral — COMMENTED** [review]
figuring out how to include fortran documentation

File: doc/api.rst [hunk 1]
```diff
@@ -1,14 +1,26 @@
...
+.. autosummary::
+   :toctree:generated/
+
+   xcape.core.calc_cape
+   xcape.core.calc_srh
+   xcape.Bunkers_model_lev
+   xcape.Bunkers_model_lev.bunkers_loop_ml
+   xcape.Bunkers_pressure_lev.bunkers_loop_pl
+   Bunkers_model_lev.bunkers_loop_ml
+..
```
[the remark this answers is not in the document]
Comment (chiaral) [reply doc/api.rst:22]: @rabernat 

https://sphinx-fortran.readthedocs.io/en/latest/user.autodoc.html#

I think I have to 
1) add sphinx-fortran to the pip list of packages in environments.yml
2) Add this extension to your sphinx conf.py in particular add 
sphinxfortran.fortran_domain
and
sphinxfortran.fortran_autodoc
to the list of extensions in conf.py

3) define all the variables as listed in https://sphinx-fortran.readthedocs.io/en/latest/user.autodoc.html#

is it worth it? does it need fortran compiler (in that case we can leave it out...)

File: src/xcape/core.py [hunk 2]
```diff
@@ -132,81 +138,83 @@ def calc_cape(*args, **kwargs):
     Vertical level options should be specified based on the input model data, whether defined on 
     pressure levels or model levels.
     
-    Formula:
-    --------
-    Calulates CAPE for a user specified set of parcel options based on the integration:
-    .. math:: \text{CAPE} = g \int_{LFC}^{EL} \frac{(\Theta_v_{parcel} - \Theta_v_{env})}{  \
-              \Theta_v_{env}} d\text{dz}
```
Comment (xebadir) [inline src/xcape/core.py:139]: Should be \text{dz} not d\text{dz} here and the subsequent line.

File: src/xcape/core.py [hunk 3]
```diff
@@ -132,81 +138,83 @@ def calc_cape(*args, **kwargs):
     Vertical level options should be specified based on the input model data, whether defined on 
     pressure levels or model levels.
     
-    Formula:
-    --------
-    Calulates CAPE for a user specified set of parcel options based on the integration:
-    .. math:: \text{CAPE} = g \int_{LFC}^{EL} \frac{(\Theta_v_{parcel} - \Theta_v_{env})}{  \
```
Comment (xebadir) [inline src/xcape/core.py:138]: Is there any way we can change the theta symbol - its pretty ugly here.

File: src/xcape/core.py [hunk 4]
```diff
@@ -132,81 +138,83 @@ def calc_cape(*args, **kwargs):
...
+        options include: 'pseudo-liquid', 'reversible-liquid','pseudo-ice', 'reversible-ice'  
+    pinc : float, optional, default is 500
         Pressure increment for integration (Pa) - Recommended usage (between 1000 and 100) is 
         based on desired speed, with accuracy of the calculation increasing with smaller 
         integration increments. 
-    method : {'fortran', 'numba'}
-        Option to select numerical approach using wrapped Fortran 90 or a Numba python variant.
-    vertical_lev : {'sigma', 'pressure'}
-        Option to select vertical grid, between model coordinates and pressure levels.
+    method : str, optional, default is 'fortran')
```
Comment (xebadir) [inline src/xcape/core.py:171]: extra bracket following 'fortran'

File: src/xcape/core.py [hunk 5]
```diff
@@ -132,81 +138,83 @@ def calc_cape(*args, **kwargs):
...
+    zMUlev : array-like
         height of MUlev (m) (only returned for source: {'most-unstable'}
-     """
+        
+    Notes
+    -----
+    CAPE is calculated on a user specified set of parcel options based on the integration:
+    
+    
+    .. math:: CAPE = g \\int_{LFC}^{EL} (\Theta_{v,parcel} - \Theta_{v,env}/(\Theta_{v,env}) d\\text{dz}
```
Comment (xebadir) [inline src/xcape/core.py:192]: Should be \text{dz} not d\text{dz} here and the subsequent line.
Can we change the theta symbol - this one is confusing with Geopotential?

File: src/xcape/core.py [hunk 6]
```diff
@@ -132,81 +138,83 @@ def calc_cape(*args, **kwargs):
...
+    -------    
+    Example of usage:
+    
+    >>> cape, cin = core.calc_cape(p, t, td, ps, ts, tds, source ='mixed-layer',
+                    mldepth=500., adiabat='pseudo-liquid', pinc = 500., 
+                    method='fortran', vertical_lev='sigma')
+   
+    References
+    ----------
+    .. [1] prova
```
Comment (xebadir) [inline src/xcape/core.py:215]: Add reference:
Emanuel, K. A. (1994). Atmospheric convection. Oxford University Press on Demand.

File: src/xcape/core.py [hunk 7]
```diff
@@ -322,6 +330,8 @@ def _calc_cape_numpy(*args,
 def calc_srh(*args, **kwargs):
     """
     Calculate storm relative helicity for a vectorized set of profiles over the first axis of the arrays.
+
```
Comment (xebadir) [inline src/xcape/core.py:333]: Did this never get put through the pull requests? I wrote a whole description for this.
    """
    Calculate Storm Relative Helicity (SRH) for a defined storm motion. 
    Description:
    ------------
    Calculate SRH over a predefined depth for a N-dimensional gridded field. 
    Performs a point profile integration of the area between the hodograph and      
    the vectors between the estimated storm motion vector at the bottom and top     
    layer. The calculation assumes that the storm motion is described by the
    un-weighted mean 0-6 km wind vector subsampled at 500m intervals, with a 
    left or right moving storm deviating by 7.5 m/s perpendicular to this 
    mean wind, following the empirically estimated approach defined by 
    Bunkers et al. (2000). Output can be specified to include the assumed 
    motion, or to return only the helicity as calculated for the user-defined 
    layer. Vertical level option should be specified based on the input model 
    data, whether defined on pressure or model levels.     
    
    Formula:
    --------
    Calculates SRH for a user specified depth based on the numeric integration:
    .. math:: \text{SRH} = \int\limits_0^top (\bar v - c) \cdot \bar\omega_{h} \,dz    
    
    * :math:'SRH' Storm Relative Helicity
    * :math:'\bar v' Environmental Wind Vector
    * :math:'\bar c' Storm Motion Vector (left and right components)
    * :math:'\bar\omega_{h}' Horizontal Vorticity Vector
    * :math:'z' height above ground
 
    For further details see Markowski and Richardson [2010], pages 230-231.
    Parameters
    ----------
    p : 'array-like'
        Atmospheric pressure in hPa.
        When vertical_lev='model', p.shape = t.shape = (nlev, x, y, ...)
        When vertical_lev='pressure', p.shape = t.shape[0] = (nlev)
    t : 'array-like'
        Atmospheric temperature in Celsius. Vertical shape should be identical to pressure. 
    td : array-like
        Atmospheric dew point temperature in Celsius. Vertical shape should be identical to pressure. 
    u  : 'array-like'
        Zonal wind component in meters per second. Vertical shape should be identical to pressure. 
    v  : 'array-like'
        Meridional wind component in meters per second. Vertical shape should be identical to pressure.    
    ps : 'array-like'
        Surface Pressure in hPa.
    ts : 'array-like'
        Surface Temperature in Celsius.
    tds : 'array-like'
        Surface dew point temperature in Celsius.
    us : 'array-like'
        10m zonal wind field in meters per second. 
    us : 'array-like'
        10m meridional wind field in meters per second. 
    
    Default usage:
    --------------
    srh_rm, srh_lm = core.calc_srh(p,t,td,u,v,ps,ts,tds,us,vs,
                                   depth=3000.,vertical_lev='sigma',
                                   output_var='srh')
    
    Optional Kwargs:
    ----------------
    The following options can be user selected.
    depth : float, optional
        Depth in meters (m) of the layer to calculate SRH.
    vertical_lev : {'sigma', 'pressure'}
        Option to select vertical grid, between model coordinates and 
        pressure levels.
    output_var : {'srh', 'all'}
        Option to return either calculated SRH only, or also the Bunker's storm
        motion for the right-moving and left-moving storms, along with the not
        pressure-weighted mean 0-6 km wind.
        'srh' = for only srh
        'all' = for srh, Bunkers' right-moving and left-moving storm component, 
                mean not pressure averaged 6km wind
    
    Returns
    -------
    srh_rm : 'array-like'
          Storm relative helicity for the right-moving storm.
    srh_lm : 'array-like'
          Storm relative helicity for the left-moving storm.
    bunkers_rm: 'array-like'
          Estimated storm motion for the right-moving storm.
    bunkers_rm: 'array-like'
          Estimated storm motion for the left-moving storm.
    bunkers_rm: 'array-like'
          Mean wind between 0 and 6km without pressure weighting.
    """

File: src/xcape/core.py [hunk 8]
```diff
@@ -381,6 +419,10 @@ def _calc_srh_gufunc(*args, **kwargs):
 # the numpy version of the algorithm
 def _calc_srh_numpy(*args,
                     depth = 3000, vertical_lev='sigma',output_var='srh'):    
+    ''' 
+    Wrapper function for srh calculation to setup optional kwargs and ensure data is
+    is provided in a format suitable output to call SRH calculation.
```
Comment (xebadir) [inline src/xcape/core.py:424]: """ 
     Wrapper function for SRH calculation to setup optional kwargs and 
     ensure data is provided in a format suitable output to call the 
     Fortran implementation of SRH calculation. Also calculates standard
     height using the hypsometric equation, which necessitates the passing
     of pressure, temperature and dewpoint temperature. Called by 
     _calc_srh_gufunc.   
    """

File: src/xcape/core.py [hunk 9]
```diff
@@ -132,81 +138,83 @@ def calc_cape(*args, **kwargs):
...
+    .. math:: CAPE = g \\int_{LFC}^{EL} (\Theta_{v,parcel} - \Theta_{v,env}/(\Theta_{v,env}) d\\text{dz}
+
+    .. math:: CIN = g \\int_{SFC}^{LFC} (\Theta_{v,parcel} - \Theta_{v,env})/(\Theta_{v,env}) d\\text{dz}
+    
+    * :math:`CAPE` = Convective available potential energy 
+    * :math:`CIN` = Convective inhibition
+    * :math:`LFC` = Level of free convection
+    * :math:`EL` = Equilibrium level
+    * :math:`g` = Gravitational acceleration
+    * :math:`\Theta_{v,parcel}` = Virtual potential temperature of the parcel
```
Comment (xebadir) [inline src/xcape/core.py:201]: If we change the theta - change here too

File: src/xcape/core.py [hunk 10]
```diff
@@ -132,81 +138,83 @@ def calc_cape(*args, **kwargs):
...
+                    method='fortran', vertical_lev='sigma')
+   
+    References
+    ----------
+    .. [1] prova
+
+    """
 
     if len(args)<6:
         raise ValueError("Too little arguments.")
```
Comment (xebadir) [inline src/xcape/core.py:220]: Do we have to use the word little? Too few is preferable - but I don't know if this is typical for python.

File: src/xcape/core.py (hunk 4 above)
Comment (chiaral) [reply src/xcape/core.py:171]: done!

File: src/xcape/core.py (hunk 5 above)
Comment (chiaral) [reply src/xcape/core.py:192]: done!

File: src/xcape/core.py (hunk 7 above)
Comment (chiaral) [reply src/xcape/core.py:333]: figuring this out. It's not on master. but it was in the PR I merged. not sure.

File: src/xcape/core.py (hunk 8 above)
Comment (chiaral) [reply src/xcape/core.py:424]: figuring this out too

File: src/xcape/core.py (hunk 7 above)
Comment (chiaral) [reply src/xcape/core.py:333]: I included it.

File: src/xcape/core.py (hunk 8 above)
Comment (chiaral) [reply src/xcape/core.py]: added

File: doc/api.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (chiaral) [reply doc/api.rst:19]: not necessary.

**Comment by chiaral** [comment]
I am going to merge this because it is getting too long

**Pull request merged**